src/main/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheck.java

```diff
@@ -0,0 +1,210 @@
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+
+/**
+ * This implementation checks that keys of properties files are sorted.
+ * It is using the binary order from String.CASE_INSENSITIVE_ORDER.compare()
+ * The sort direction is ascending.
+ *
+ *   <p>
+ *       Example properties file:
+ *   </p>
+ *   <pre>
+ *       &#64;=64 (it is the 'at' sign)
+ *       A=65
+ *       a=97
+ *       &auml;=132 - violation
+ *       key=107 than nothing
+ *       key.sub=k is 107 and dot is 46
+ *   </pre>
+ *   <ul>
+ *       <li>
+ *           The &#64; sign is on position 64 and comes first.
+ *       </li>
+ *       <li>
+ *           The capital A is on 65 and the lowercase a is on position 97 on the ascii table.
+ *       </li>
+ *       <li>
+ *           On line 4 you have an a umlaut. This belongs to the extended ascii table and comes at
+ *           last position. This is the error here.
+ *       </li>
+ *       <li>
+ *           Key and subkey are in correct order here, because only keys are relevant.
+ *           Therefore on line 5 you have only "key" an nothing behind.
+ *           On line 6 you have "key." The dot is on position 46 which is higher than nothing.
+ *       </li>
+ *   </ul>
+ */
+@StatelessCheck
+public class OrderedPropertiesCheck extends AbstractFileSetCheck {
+
+    /**
+     * Localization key for check violation.
+     */
+    public static final String MSG_KEY = "properties.notSorted.property";
+    /**
+     * Localization key for IO exception occurred on file open.
+     */
+    public static final String UNABLE_TO_OPEN_FILE = "Unable to open file: ";
+    /**
+     * Pattern matching single space.
+     */
+    private static final Pattern SPACE_PATTERN = Pattern.compile(" ");
+
+    /**
+     * Construct the check with default values.
+     */
+    public OrderedPropertiesCheck() {
+        setFileExtensions("properties");
+    }
+
+    /**
+     * Processes the file and check order.
+     * @param file the file to be processed
+     * @param fileText the contents of the file.
+     * @throws CheckstyleException
+     * I like to keep the order - keySet is not possible
+     * @noinspection EnumerationCanBeIteration
+     */
+    @Override
+    protected void processFiltered(File file, FileText fileText) throws CheckstyleException {
+        final SequencedProperties properties = new SequencedProperties();
+        try (InputStream inputStream = Files.newInputStream(file.toPath())) {
+            properties.load(inputStream);
+        }
+        catch (IOException ex) {
+            throw new CheckstyleException(UNABLE_TO_OPEN_FILE + file.getPath(), ex);
+        }
+
+        String previousProp = "";
+
+        final Enumeration<Object> keys = properties.keys();
+
+        while (keys.hasMoreElements()) {
+
+            final String propKey = (String) keys.nextElement();
+
+            //if (previousProp.compareTo(propKey) > 0) {
```

Comment:
This item has not been done from https://github.com/checkstyle/checkstyle/pull/6343#discussion_r245649833 .
> Please remove commented code.

Reply:
OK, I will. My plan was to allow case sensitive an case insensitive check. But I think I will postpone it, because this MR getting too long to fulfil.

Reply:
deleted. Done